utils/getUserLocation.js

```diff
@@ -0,0 +1,29 @@
+import { Platform, Text, View, StyleSheet } from 'react-native';
+import Device from 'expo-device';
+import * as Location from 'expo-location';
+
+
+async function getUserLocation () {
+    console.log('in function');
+      if (Platform.OS === 'android' && !Device.isDevice) {
+        setErrorMsg(
```

Comment:
where is this function defined? I can't see where it would be used

Reply:
Thats the declaration. Its called in the Home.jsx

Reply:
the setErrorMsg isn't defined anywhere - sorry should have been more specific

Reply:
If I remove the isDevice error, and fix the catch err, then it flags that setErrorMsg is undefined too

Reply:
Just looking in the docs and saw this which I think fixes the device error: import * as Device from 'expo-device';